### ISSUE TITLE
`statement.setObject(index, null);` not compatible with Exasol

### ISSUE THREAD

**Issue by smaspe** [confluentinc/kafka-connect-jdbc#851]
## TL;DR
 `statement.setObject(index, null);` is not supported by all databases and crashes when used with batches and a null Timestamp in Exasol.

## Issue

In [`GenericDatabaseDialect.bindField()`](https://github.com/confluentinc/kafka-connect-jdbc/blob/v5.5.0/src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java#L1472-L1473), `statement.setObject(index, null);` is called.

This causes a crash in at least one particular situation when used with Exasol. (Specifically, when the records contain a Timestamp, and the first Timestamp of the batch is null).

This call to `setObject` (with null) is explicitly recommended against in [`PreparedStatement` javadoc](https://docs.oracle.com/javase/10/docs/api/java/sql/PreparedStatement.html#setNull(int,int)):

> **Note:** Not all databases allow for a non-typed Null to be sent to the backend. For maximum portability, the `setNull` or the `setObject(int parameterIndex, Object x, int sqlType)` method should be used instead of `setObject(int parameterIndex, Object x)`.

## Reproduction

Any batch processing that contains a null value in one of the records will go through that method.

How it triggers a particular issue with Exasol, (if relevant) is very simple. Given any `PreparedStatement` from a `jdbc:exa` connection:

```
statement.setObject(1, null);
statement.addBatch();
statement.setTimestamp(1, new Timestamp(42), Calendar.getInstance());
```

## Fixing

The recommendation from `PreparedStatement` javadoc is to provide a `sqlType`. The `sqlType` can be obtained by looking at the schema:

```
private int getSqlType(Schema schema) {
  if (schema.name() != null) {
    switch (schema.name()) {
      case Date.LOGICAL_NAME:
        return Types.DATE;
      // ...
    }
  }
  switch (schema.type()) {
    case INT8:
      return Types.TINYINT;
    // ...
  }
  throw new ConnectException("Unsupported source data type: " + schema.type());
}
```
